File: web/android/player-core/src/main/java/com/jerocine/player/PlayerConfig.java
package com.jerocine.player;

import java.io.File;

/**
 * 播放器配置 — 壳层传入
 */
public class PlayerConfig {
    public String sourcesJson;
    public String currentSourceId;
    public int startIndex;
    public long resumeMs;
    public long skipIntroMs;
    public long skipOutroMs;
    public String filmId;
    public String filmName;
    public String proxyBase;
    public File cacheDir;

    public PlayerConfig(
        String sourcesJson,
        String currentSourceId,
        int startIndex,
        long resumeMs,
        long skipIntroMs,
        long skipOutroMs,
        String filmId,
        String filmName,
        String proxyBase,
        File cacheDir
    ) {
        this.sourcesJson = sourcesJson;
        this.currentSourceId = currentSourceId;
        this.startIndex = startIndex;
        this.resumeMs = resumeMs;
        this.skipIntroMs = skipIntroMs;
        this.skipOutroMs = skipOutroMs;
        this.filmId = filmId;
        this.filmName = filmName;
        this.proxyBase = proxyBase;
        this.cacheDir = cacheDir;
    }
}

File: android/player-core/src/main/java/com/jerocine/player/PlayerConfig.kt
package com.jerocine.player

import java.io.File

/**
 * 播放器配置 — 壳层传入
 */
data class PlayerConfig(
    val sourcesJson: String,       // 多源 JSON
    val currentSourceId: String, // 当前源 ID
    val startIndex: Int,         // 起始集索引
    val resumeMs: Long,          // 续播位置
    val skipIntroMs: Long,       // 跳片头
    val skipOutroMs: Long,       // 跳片尾
    val filmId: String,          // 影片 ID
    val filmName: String,        // 影片名
    val proxyBase: String,       // 代理 base
    val cacheDir: File,          // 缓存目录（壳层传，各自独立）
)
